### PR TITLE
docs: update the docs about extract css

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | update the docs about extract css in ssr case  |
| 🇨🇳 Chinese | 更新 SSR 案例中提取 CSS 的文档 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 84ab64b</samp>

Update the documentation of customizing theme with cssinjs in both English and Chinese. Replace the deprecated `static-style-extract` script with the new `cssinjs` and `doExtraStyle` script. Improve the performance and compatibility of the theme customization feature.

### <samp>🤖 Generated by Copilot at 84ab64b</samp>

* Replace the deprecated `static-style-extract` script with `cssinjs` and `doExtraStyle` script in the documentation of customizing theme with cssinjs ([link](https://github.com/ant-design/ant-design/pull/41686/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L316-R415), [link](https://github.com/ant-design/ant-design/pull/41686/files?diff=unified&w=0#diff-7238b66024192aa79bff8e41259edd3a2c30155c7422b58c53e6fc63d679a4afL312-R411))

### CODE REVIEW

File: docs/react/customize-theme.zh-CN.md [hunk 1]
```diff
@@ -309,160 +309,105 @@ npm install ts-node tslib --save-dev
...
-
-如果你的项目中使用了自定义主题，可以尝试通过以下方式进行烘焙：
-
-```tsx
-import { extractStyle } from '@ant-design/static-style-extract';
-import { ConfigProvider } from 'antd';
+import { StyleProvider, createCache } from '@ant-design/cssinjs';
+import Document, { DocumentContext, Head, Html, Main, NextScript } from 'next/document';
+import { doExtraStyle } from '../scripts/genAntdCss';
+export default class MyDocument extends Document {
```
Comment (afc163) [inline docs/react/customize-theme.zh-CN.md:362]: 1. 少空行
2. 别用 class component 吧

Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md:362]: 1. 我改一下
2. 这是 Next.js 的默认使用方式吧
image

File: docs/react/customize-theme.en-US.md [hunk 2]
```diff
@@ -313,160 +313,106 @@ npm install ts-node tslib --save-dev
...
 // scripts/genAntdCss.tsx
-import { extractStyle } from '@ant-design/static-style-extract';
+import { extractStyle } from '@ant-design/cssinjs';
+import type Entity from '@ant-design/cssinjs/lib/Cache';
+import { createHash } from 'crypto';
 import fs from 'fs';
+import path from 'path';
 
-const outputPath = './public/antd.min.css';
+const styleTagReg = /<style[^>]*>([\s\S]*?)<\/style>/g;
```
Comment (afc163) [inline docs/react/customize-theme.en-US.md:322]: 正则匹配会不会有点黑？

Comment (kiner-tang) [reply docs/react/customize-theme.en-US.md:322]: 那咱们是不是要在 cssinjs 当中提供导出时是否需要 <style> 标签的选项呢？目前好像都会带上 <style>标签

Comment (zombieJ) [reply docs/react/customize-theme.en-US.md:322]: 可以的，本来它就是包一层。加个 param 相当于不要包，还省性能

Comment (kiner-tang) [reply docs/react/customize-theme.en-US.md:322]: image
类似这样，觉得靠谱不？

Comment (zombieJ) [reply docs/react/customize-theme.en-US.md:322]: 嗯，可以的

Comment (kiner-tang) [reply docs/react/customize-theme.en-US.md:322]: 好的，那我搞一下

Comment (kiner-tang) [reply docs/react/customize-theme.en-US.md:322]: - https://github.com/ant-design/cssinjs/pull/103

File: docs/react/customize-theme.zh-CN.md [hunk 3]
```diff
@@ -309,160 +309,106 @@ npm install ts-node tslib --save-dev
...
 // scripts/genAntdCss.tsx
-import { extractStyle } from '@ant-design/static-style-extract';
+import { extractStyle } from '@ant-design/cssinjs';
+import type Entity from '@ant-design/cssinjs/lib/Cache';
+import { createHash } from 'crypto';
 import fs from 'fs';
+import path from 'path';
 
-const outputPath = './public/antd.min.css';
+const styleTagReg = /<style[^>]*>([\s\S]*?)<\/style>/g;
```
Comment (zombieJ) [inline docs/react/customize-theme.zh-CN.md:318]: 这个应该不需要了~

**Comment by MadCcc** [comment]
两种 extract 的方式最好都提一下，链接也都给上

**Review by MadCcc — APPROVED** [review]

**Comment by afc163** [comment]
@kiner-tang 这个文档可以单独提一篇，现在放在定制主题里（用户大多数是用 next.js 遇到问题，不一定是需要定制主题），心智貌似不太对，不太好找。

**Pull request merged**